shared/src/main/scala/coop/rchain/catscontrib/BooleanF.scala

```diff
@@ -0,0 +1,27 @@
+package coop.rchain.catscontrib
+
+import cats._, cats.data._, cats.implicits._
+
+final class BooleanF[F[_]: Applicative](val self: F[Boolean]) {
+  def &&^(other: F[Boolean]): F[Boolean] =
+    BooleanF.&&^[F](self, other)
+  def ||^(other: F[Boolean]): F[Boolean] =
+    BooleanF.&&^[F](self, other)
+  def ~^(): F[Boolean] = BooleanF.~^[F](self)
+}
+
+object BooleanF {
+  def &&^[F[_]: Applicative](m1: F[Boolean], m2: F[Boolean]): F[Boolean] =
+    Applicative[F].map2(m1, m2)(_ && _)
```

Comment:
I've meant this object - it's unused AFAIR

Reply:
I've used it in this PR :)

Reply:
oh... you mean that there is both object and implict class that uses the object? that is style i've seen in both cats and scalaz and i've been using that ever since. you give API with methods that can be used in both was: as a function as a method on an object

Reply:
You've used it in this very file only. It's effectively-unused and has API I'd rather not use anywhere else than this file.

`&&^(foo, &&^(bar, baz))` ?!

Reply:
I'm OK if it's private. I'm ok if it's public too, it's not that bad - unless somebody nests calls like me above :)